test/unit/processing/components/get-schedule-dates.test.js

```diff
@@ -93,7 +93,12 @@ describe('get schedule dates', () => {
 
```

Comment:
I find this test file v difficult to quantify it's completeness simply because it is difficult to read for test cases.

We have discussed in our dev chat around using `describe` more effectively to better split up our test files when they cover multiple test cases -- I think this is a great contender for tht!

from skimming over it, I can see the following scenarios:
1. all segments are outstanding
2. all segment are not outstanding
3. delta >0
4. delta <0
5. delta 0 (as discussed below, this isn't covered)
6. if top-up
7. if reduction
8. if recovery

From v quickly going through and not understanding the code entirely.

Could we use `describe` to better split it up (I think we'd both agree that this function needs splitting it and covered with my granular unit tests (tech debt ticket, pls) but in the meantime, this might be our best bet)

Sommit like:

describe('get schedule dates') <-- maybe also there is a better name for this?
beforeEach() <-- common to ALL tests, likely setting up our objects
afterEach() <-- common to ALL tests, usually clearing mocks
describe('if delta is greater than zero')
beforeEach (setting delta >0)
describe('if all payments are outstanding')
test, test test
describe('if all payments are not outstanding')
test, test, test
etc
etc

This way, just for this simple example we can see our test scenarios come out:

- when delta is >0 
and all payments are outstanding expect ...
and all payments are not outstanding expect ...
etc
- when delta is <0
and all payments are outstanding expect ...
and all payments are not outstanding expect ...
etc

Reply:
Added as tech-debt ticket